src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,261 @@
+        Map<String, String> dataMap = getMessageHeadersAsMap(message);
+        dataMap.put("stack_trace", Arrays.toString(exception.getStackTrace()));
+        LOGGER.error(
+                String.format("order-received message processing failed with a non-recoverable exception. %n%1$s",
+                        dataMap.toString())
+        );
+    }
 
-        config.setRoundRobinPartitioner(true);
-        config.setAcks(Acks.WAIT_FOR_ALL);
-        config.setRetries(10);
-        return config;
+    private void logMessageProcessed(org.springframework.messaging.Message<OrderReceived> message){
+        LOGGER.info(String.format("Order received message successfully processed. %1$s",
+                getMessageHeadersAsMap(message).toString()));
     }
 
-    protected void republishMessageToTopic(String orderUri, String currentTopic, String nextTopic, Exception ex)
+    private Map<String, String> getMessageHeadersAsMap(org.springframework.messaging.Message<OrderReceived> message){
+        Map<String, String> dataMap = new HashMap<>();
+        MessageHeaders messageHeaders = message.getHeaders();
+        dataMap.put("data.key", "" + messageHeaders.get(KafkaHeaders.RECEIVED_MESSAGE_KEY));
+        dataMap.put("data.topic", "" + messageHeaders.get(KafkaHeaders.RECEIVED_TOPIC));
+        dataMap.put("data.offset", "" + messageHeaders.get(KafkaHeaders.OFFSET));
+        dataMap.put("data.partition", "" + messageHeaders.get(KafkaHeaders.RECEIVED_PARTITION_ID));
+
+        return dataMap;
+    }
+
+    protected void republishMessageToTopic(String orderUri, String currentTopic, String nextTopic)
             throws SerializationException, ExecutionException, InterruptedException {
-        LOGGER.error("Processing message: " + orderUri + " received on topic: " + currentTopic
-                + " failed with exception: " + ex.getStackTrace());
-        Message message = createRetryMessage(orderUri);
-        LOGGER.info("Republishing message: " + orderUri + " received on topic: " + currentTopic
-                + " to topic: " + nextTopic);
-        for (int attempt = 0; attempt < MAX_RETRY_ATTEMPTS; attempt++) {
-            if (currentTopic.equals(ORDER_RECEIVED_TOPIC)) {
-                chKafkaConsumerGroupMain.retry(attempt, message);
-            }
-            else if (currentTopic.equals(ORDER_RECEIVED_TOPIC_RETRY)) {
-                chKafkaConsumerGroupRetry.retry(attempt, message);
-            }
-        }
+        LOGGER.info(String.format("Republishing message: \"%1$s\" received from topic: \"%2$s\" to topic: \"%3$s\"",
+                orderUri, currentTopic, nextTopic));
+        Message message = createRetryMessage(orderUri, nextTopic);
+        kafkaProducer.sendMessage(message);
     }
 
-    protected Message createRetryMessage(String orderUri) throws SerializationException {
+    protected Message createRetryMessage(String orderUri, String topic) throws SerializationException {
```

Comment:
Has a story been raised for handling an attempt field?

Reply:
`@Retryable` manages the count of attempts internally, so it doesn't need to be explicitly managed/handled in this implementation.

Reply:
There is now a `retryCount` field that keeps track of retries for each unsuccessful message received from each topic.